src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -30,15 +31,14 @@ using PyRTMap = ov::RTMap;
 
 PYBIND11_MAKE_OPAQUE(PyRTMap);
 
-static void set_tensor_names(const ov::ParameterVector& parameters) {
-    for (const auto& param : parameters) {
-        ov::Output<ov::Node> p = param;
-        if (p.get_node()->output(0).get_names().empty()) {
-            std::unordered_set<std::string> p_names({p.get_node()->get_friendly_name()});
-            p.get_node()->output(0).set_names(p_names);
-        }
-    }
+namespace {
+template <class... Args>
+std::shared_ptr<ov::Model> make_model_with_tensor_names(Args&&... args) {
+    auto model = std::make_shared<ov::Model>(std::forward<Args>(args)...);
+    ov::util::set_tensors_names(ov::AUTO, *model);
```

Comment:
Am I right that each output node will have a name?
I am asking in the context of the issue in js api, when `output.get_any_name()` fails.
Or here are also problems with resolving names collisions

Reply:
Is added only for python API, only model input and outputs will have tensor names when created with python API.
Similar approach can be used in other binding, but there is plan to add in C++ API, but now is not possible yet